src/main/java/internet/market/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,105 @@
+import java.util.Optional;
+
+public class ProductDaoJdbcImpl implements ProductDao {
+
+    @Override
+    public Product create(Product item) {
+        String query = "INSERT INTO products (name, price) VALUES (?, ?) ";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection
+                    .prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            statement.setString(1, item.getName());
+            statement.setDouble(2, item.getPrice());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                item.setId(resultSet.getLong(1));
+            }
+            return item;
+        } catch (SQLException e) {
+            throw new DataProcessingException("We can`t add product with id = " + item.getId(), e);
+        }
+    }
+
+    @Override
+    public Optional<Product> get(Long id) {
+        String query = "SELECT * FROM products WHERE deleted=false AND product_id=?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getProductInformationFromDataBase(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t find product with id = " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Product> getAll() {
+        String query = "SELECT * FROM products WHERE deleted=false";
+        List<Product> productList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                productList.add(getProductInformationFromDataBase(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get all products", e);
+        }
+        return productList;
+    }
+
+    @Override
+    public Product update(Product item) {
+        String query = "UPDATE products SET name=?, price=? WHERE product_id=?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, item.getName());
+            statement.setDouble(2, item.getPrice());
+            statement.setLong(3, item.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t add product with id = " + item.getId(), e);
```

Comment:
do we add product here, or we update it ?

Reply:
done